src/encoder.rs

```diff
@@ -127,14 +127,40 @@ fn encode_arg(arg: &OscType) -> Result<(Option<Vec<u8>>, char)> {
             for (i, v) in x.iter().enumerate() {
                 bytes[i + 4] = *v;
             }
-            Ok((Some(bytes), 'b'))
+            Ok((Some(bytes), 'b'.to_string()))
         }
-        OscType::Time(ref x, ref y) => Ok((Some(encode_time_tag(*x, *y)), 't')),
-        OscType::Midi(ref x) => Ok((Some(vec![x.port, x.status, x.data1, x.data2]), 'm')),
-        OscType::Color(ref x) => Ok((Some(vec![x.red, x.green, x.blue, x.alpha]), 'r')),
-        OscType::Bool(ref x) => if *x { Ok((None, 'T')) } else { Ok((None, 'F')) },
-        OscType::Nil => Ok((None, 'N')),
-        OscType::Inf => Ok((None, 'I')),
+        OscType::Array(ref x) => {
```

Comment:
I would like to move the `Array` case to the bottom since its the most complex one.